src/VizTab.tsx

```diff
@@ -1,29 +1,121 @@
+    {namespace: string, x: number, y: number, theta: number}) {
+  const fbb = new flatbuffers.Builder();
+  const pose = fb.amrl_msgs.Pose2Df.createPose2Df(
+    fbb,
+    fb.MsgMetadata.createMsgMetadata(
+      fbb,
+      fbb.createString("amrl_msgs/Pose2Df"),
+      fbb.createString(`${namespace}/move_base_simple/goal`)
+    ),
+    x,
+    y,
+    theta
+  );
+  fbb.finish(pose);
+  return fbb.asUint8Array();
+}
+
+// oof
+// not sure why we don't get createX() methods for some of these
+function createLocalization2DMsg({namespace, frame="map", map, x, y, theta}:
+    {namespace: string, frame?: string, map: string, x: number, y: number, theta: number}) {
+  const fbb = new flatbuffers.Builder();
+
+  const metadataOffset = fb.MsgMetadata.createMsgMetadata(
+    fbb,
+    fbb.createString("amrl_msgs/Localization2DMsg"),
+    fbb.createString(`${namespace}/initialpose`)
+  );
+
+  const frameOffset = fbb.createString(frame);
+
+  fb.std_msgs.Header.startHeader(fbb);
+  fb.std_msgs.Header.addStamp(fbb, fb.RosTime.createRosTime(
+    fbb,
+    Math.floor(Date.now() / 1000),
+    0
+  ));
+
+  fb.std_msgs.Header.addFrameId(fbb, frameOffset);
+  const headerOffset = fb.std_msgs.Header.endHeader(fbb);
+
+  const mapOffset = fbb.createString(map);
+
+  const poseOffset = fb.amrl_msgs.Pose2Df.createPose2Df(
+    fbb,
+    0,
+    x,
+    y,
+    theta
+  );
+  fb.amrl_msgs.Localization2DMsg.startLocalization2DMsg(fbb);
+  fb.amrl_msgs.Localization2DMsg.add_Metadata(fbb, metadataOffset);
+  fb.amrl_msgs.Localization2DMsg.addHeader(fbb, headerOffset);
+  fb.amrl_msgs.Localization2DMsg.addMap(fbb, mapOffset);
+  fb.amrl_msgs.Localization2DMsg.addPose(fbb, poseOffset);
+  const loc = fb.amrl_msgs.Localization2DMsg.endLocalization2DMsg(fbb);
+
+  fbb.finish(loc);
+  return fbb.asUint8Array();
+}
+
 function Alert(props: AlertProps) {
   return <MuiAlert elevation={6} variant="standard" {...props}/>;
 }
 
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    Default: {
+      cursor: 'default'
+    },
+    Localize: {
+      cursor: 'crosshair'
+    },
+    SetNav: {
+      cursor: 'pointer'
+    },
+  })
+);
+
 export default function VizTab(props: {namespace: string}) {
+  type ClickAction = "Default" | "Localize" | "SetNav";
+
   const ws = useContext(WebSocketContext);
+  const canvasUtils = useRef<CanvasUtilsRef>(null);
+  const classes = useStyles();
+
   const [locAlertOpen, setLocAlertOpen] = useState(true);
   const [baseLink, setBaseLink] = useState(new THREE.Matrix4());
 
+  const [clickAction, setClickAction] = useState<ClickAction>("Default");
+  const rc = useContext(RobotContext);
+  
+  const [locShowMap, setLocShowMap] = useStorage("Localization.showMap", true);
+  const [scanShow, setScanShow] = useStorage("LaserScan.show", true);
+  const [vizShowPoints, setVizShowPoints] = useStorage("Viz.showPoints", false);
+  const [vizShowLines, setVizShowLines] = useStorage("Viz.showLines", false);
+
```

Comment:
could store `mapName` state here. May or may not want to persist it to storage... not sure how that would play out in real usage.